recipes/icu/all/conanfile.py

```diff
@@ -72,15 +79,18 @@ def build(self):
             tools.replace_in_file(run_configure_icu_file, "-MD", flags)
 
         self._workaround_icu_20545()
+        self._remove_scrptrun()
 
         self._env_build = AutoToolsBuildEnvironment(self)
         if not self.options.get_safe("shared"):
             self._env_build.defines.append("U_STATIC_IMPLEMENTATION")
         if tools.is_apple_os(self.settings.os):
             self._env_build.defines.append("_DARWIN_C_SOURCE")
             if self.settings.get_safe("os.version"):
-                self._env_build.flags.append(tools.apple_deployment_target_flag(self.settings.os,
-                                                                            self.settings.os.version))
+                self._env_build.flags.append(tools.apple_deployment_target_flag(self.settings.os, self.settings.os.version))
+
+        if self._is_msvc:
+            self._env_build.cxx_flags.append("-EHsc")
```

Comment:
why do we need this now?

Reply:
Not my intention, maybe something that was there when I created this PR. I've removed it.

Reply:
It's not ABI breaking, I think.
See https://docs.microsoft.com/en-us/cpp/build/reference/eh-exception-handling-model?view=vs-2019

Defining `/EHsc` is an optimization as it assumes that c code will not throw an exception
This is fine as long as the dependencies do not include any c++ libraries.
It looks like Microsoft advises to use `/EHsc`.

Reply:
I think it is, as `extern "C"` functions may potentially throw exceptions (at least C++ standard doesn't forbid it), and you may catch them (from the C++ code, ofcourse), but using `/EHsc` changes that behavior and breaks the contract.

Reply:
You might be right.
It looks like it only matters if you're doing `catch(...)` or using SEH.